src/prpy/futures.py

```diff
@@ -167,7 +167,7 @@ def add_done_callback(self, fn):
         @type  fn: (ResultType) -> None
         """
         with self.lock:
-            if self._is_done:
+            if not self._is_done:
                 if fn in self._callbacks:
```

Comment:
Should this check be moved outside this `if` block entirely? it seems like it applies to both cases.